vdr/types/common.go

```diff
@@ -57,7 +58,19 @@ var ErrServiceNotFound = errors.New("service not found in DID Document")
 var ErrServiceReferenceToDeep = errors.New("service references are nested to deeply before resolving to a non-reference")
 
 // ErrInvalidServiceQuery is returned when a compound service contains an invalid service reference.
-var ErrInvalidServiceQuery = errors.New("service query is invalid")
+type ErrInvalidServiceQuery struct {
+	Cause error
+}
+
+// Error returns the error message.
+func (e ErrInvalidServiceQuery) Error() string {
+	return fmt.Sprintf("service query is invalid: %s", e.Cause)
+}
+
+// Is checks whether the other error is also a ErrInvalidServiceQuery
+func (e ErrInvalidServiceQuery) Is(other error) bool {
+	return fmt.Sprintf("%T", e) == fmt.Sprintf("%T", other)
```

Comment:
So a wrapped ErrInvalidServiceQuery will not be equal

Reply:
No, it isn't. Primary goal of this is to keep supporting the error to status code mapping in our APIs.